### PR TITLE
[rubberband] Add new port

### LINKED ISSUE #17768: [New Port Request] rubberband
Library name: rubberband

Library description: High quality software library for audio time-stretching and pitch-shifting.

Source repository URL: https://hg.sr.ht/~breakfastquay/rubberband

Project homepage (if different from the source repository): https://www.breakfastquay.com/rubberband/

I would be happy to create a draft pull request with my portfile, which probably still needs a bit of work to not accidentally pick up system libraries.

### PR DESCRIPTION
**Pull request by dreuter**
[rubberband](https://www.breakfastquay.com/rubberband/) is a small
library for stretching sounds.
This is the minimal portfile to make vcpkg happy, but probably still
needs some improvement, i.e. not picking up system libraries.

I do have some questions on how to improve this port (BTW: I am not familiar with meson and this is my first port):
- Should I delete the binaries (the rubberband "example" executable), as done in this port, or would it be better to patch the meson build file to disable building the executable in the first place?
- rubberband can use different FFT implementations. I have read that features would not be a good solution to handle this, as these would not  be strictly additive. Should I just pick one? The project itself uses a different one on macOS than all other platforms by default. Should this be reflected in this port?
- rubberband does have a VAMP plugin which is optional and in my opinion not as interesting as the library itself (in the context of vcpkg), do I need to provide a feature for that? Or is this something, that could be done in a later PR, if somebody would even be interested in it.
- Do I need to take extra steps in order to avoid picking up system libraries during the build of this package or would that be fine?

**Describe the pull request**

- #### What does your PR fix?  
  Fixes #17768 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)? 
  I have so far only tested it on my machine (linux). No I have not updated the CI baseline.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  To the best of my knowledge ;)

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  I am still working on this PR

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**
Opened it as a draft PR.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The failures on uwp

```
   Creating library librubberband.so.2.1.lib and object librubberband.so.2.1.exp
MSVCRTD.lib(utility_app.obj) : error LNK2019: unresolved external symbol __imp_RoInitialize referenced in function __scrt_initialize_winrt
librubberband.so.2.1.3 : fatal error LNK1120: 1 unresolved externals
```

**Comment by dreuter** [comment]
This seems to be related https://github.com/mesonbuild/meson/pull/4485

Should I just patch the library or disable uwp builds until the above mentioned PR is merged and released?

**Comment by NancyLi1013** [comment]
Since this is a bug in `meson`, I think we can disable uwp now until the fix has been merged in `meson`. 
cc @Neumann-A do you have any suggestions about this?

File: ports/rubberband/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+vcpkg_install_meson()
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+  file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
+
+file(
+  INSTALL ${SOURCE_PATH}/COPYING
+  DESTINATION ${CURRENT_PACKAGES_DIR}/share/rubberband
```
Comment (NancyLi1013) [inline ports/rubberband/portfile.cmake:22]:
```suggestion
  DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by daschuer** [comment]
For Mixxx we use an overlay port for Rubberband 
That can be found here:
https://github.com/mixxxdj/vcpkg/tree/2.3/overlay/ports/rubberband
It supports Windows Linux and MacOS

We have developed a CMake build system for Rubberband that lives here: 
https://github.com/Be-ing/rubberband/commit/3abe9d2ce8ada8b1d087cfc9d6f2f5199c727e14

Unfortunately it was rejected in favor of mason here: 
https://github.com/breakfastquay/rubberband/pull/18 

Maybe this info is helpful here. 

I would be happy If we can finally drop our overlay once this one is merged.

**Comment by JonLiu1993** [comment]
@dreuter ,Is this PR being worked on?

**Comment by dreuter** [comment]
> @dreuter ,Is this PR being worked on?

Hi, sorry I was working on different projects. I will try to get this finished as soon as possible, but you can also close it if you want and I will reopen it, once I've done my work.

**Comment by JonLiu1993** [comment]
@dreuter ,I’m going to close this PR. If this is something you’d like to see merged and work is being done, please reopen.

**Pull request closed without merging**